pkg/udf/forward/options.go

```diff
@@ -36,23 +37,27 @@ type options struct {
 	retryInterval time.Duration
 	// logger is used to pass the logger variable
 	logger *zap.SugaredLogger
-	// enableMapUdfStream indicates whether the message streaming is enabled or not for map UDF processing
-	enableMapUdfStream bool
 	// cbPublisher is the callback publisher for the vertex.
 	cbPublisher *callback.Uploader
-	// enableBatchMapUdf indicates whether the batch map mode is enabled for the map UDF
-	enableBatchMapUdf bool
+	// unaryMapUdfApplier is the UDF applier for a unary map mode
+	unaryMapUdfApplier applier.MapApplier
+	// streamMapUdfApplier is the UDF applier for a server streaming map mode
+	streamMapUdfApplier applier.MapStreamApplier
+	// batchMapUdfApplier is the UDF applier for a batch map mode
+	batchMapUdfApplier applier.BatchMapApplier
```

Comment:
Now that we have appliers in options for both source transformer and map udf, should we apply the same for reduce?

Reply:
We can do that for consistency!

Reply:
Do we plan to do that as part of this PR or separately?

Reply:
Separate PR is better for that